src/foam/u2/search/TextSearchView.js

```diff
@@ -100,11 +108,27 @@ foam.CLASS({
       name: 'updateValue',
       code: function() {
         var value = this.view.data;
-        this.predicate = ! value ?
-            this.True.create() :
-            this.richSearch ?
-                this.queryParser.parseString(value) :
-                this.ContainsIC(this.property, value);
+
+        if ( ! value ) {
+          this.predicate = this.True.create();
+          return;
+        }
+
+        if ( this.richSearch ) {
+          this.predicate = this.filterController.filters
+            .map((name) => this.of.getAxiomByName(name))
```

Comment:
Better to do getAxiomByClass(foam.core.String)

Reply:
If we did that, then it could return properties that aren't set as search columns on the model. The current way avoids that problem. We could do getAxiomByClass and filter afterwards to make sure the properties are search columns, but I don't see the advantage of doing it that way over the current approach.

Reply:
Where/how is it filtering search columns? Search columns are those that appear in the reciprocal search panel but it doesn't mean we can't search them in the query parser or in your full-text search.

Reply:
> Where/how is it filtering search columns? 

`this.filterController.filters` is an array of the search column names. It gets them from ReciprocalSearch, which exports itself as filterController.

> Search columns are those that appear in the reciprocal search panel but it doesn't mean we can't search them in the query parser or in your full-text search.

That's true, but I think that would seem like strange behaviour from the user's perspective. If I got rows that didn't visibly contain the text I searched for I'd be confused.